packages/siri-sx-generator/util/index.ts

```diff
@@ -0,0 +1,166 @@
+
+export const getAffectedModesList = (consequences: ApiConsequence[]) =>
+    consequences
+        .map((c) => c.vehicleMode)
+        .filter((item, index, array) => array.indexOf(item) === index)
+        .join(";");
+
+export const getAffectedOperatorsList = (consequences: ApiConsequence[]) =>
+    consequences
+        .flatMap((c) => {
+            if (isOperatorConsequence(c)) {
+                return c.consequenceOperators.flatMap((co) => co.operatorNoc);
+            }
+
+            return null;
+        })
+        .filter((item, index, array) => notEmpty(item) && array.indexOf(item) === index)
+        .join(";");
+
+export const getAffectedServicesCount = (consequences: ApiConsequence[]) =>
+    consequences
+        .flatMap((c) => {
+            if (isServicesConsequence(c)) {
+                return c.services;
+            }
+
+            return null;
+        })
+        .filter(
+            (service, index, array) =>
+                notEmpty(service) && array.findIndex((item) => item?.id === service.id) === index,
+        ).length || "";
+
+export const getAffectedStopsCount = (consequences: ApiConsequence[]) =>
+    consequences
+        .flatMap((c) => {
+            if (isServicesConsequence(c) || isStopsConsequence(c)) {
+                return c.stops;
+            }
+
+            return null;
+        })
+        .filter(
+            (stop, index, array) =>
+                notEmpty(stop) && array.findIndex((item) => item?.atcoCode === stop.atcoCode) === index,
+        ).length || "";
+
+export const convertToCsv = async (disruptions: ApiDisruption[]) => {
+    const csvDisruptions = disruptions.map((disruption) => {
+        return {
+            ...disruption,
+            validityStart: getDatetimeFromDateAndTime(
+                disruption.validity[0].disruptionStartDate,
+                disruption.validity[0].disruptionStartTime,
+            ).toISOString(),
+            validityEnd: getSortedDisruptionFinalEndDate(disruption)?.toISOString() ?? "",
+            publicationStart: getDatetimeFromDateAndTime(
+                disruption.publishStartDate,
+                disruption.publishStartTime,
+            ).toISOString(),
+            publicationEnd:
+                disruption.publishEndDate && disruption.publishEndTime
+                    ? getDatetimeFromDateAndTime(disruption.publishEndDate, disruption.publishEndTime).toISOString()
+                    : "",
+            planned: disruption.disruptionType === "planned" ? "true" : "false",
+            modesAffected: getAffectedModesList(disruption.consequences),
+            operatorsAffected: getAffectedOperatorsList(disruption.consequences),
+            servicesAffected: getAffectedServicesCount(disruption.consequences),
+            stopsAffected: getAffectedStopsCount(disruption.consequences),
+        };
+    });
+
+    return json2csv(csvDisruptions, {
```

Comment:
Have you done an `npm install`? If so it might be worth clearing down your node modules and reinstalling everything, or stopping and starting your ide